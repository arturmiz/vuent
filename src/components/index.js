import './Vnt.vue';
export { default as VntAutosuggest } from './autosuggest';
export { default as VntButton } from './button';
export { default as VntCheckbox } from './checkbox';
export { default as VntDialog } from './dialog';
export { default as VntHeader } from './header';
export { default as VntIcon } from './icon';
export { default as VntInput } from './input';
export { default as VntLink } from './link';
export { default as VntListView } from './listview';
export { default as VntNavView } from './navview';
export { default as VntPivot } from './pivot';
export { default as VntRadio } from './radio';
export { default as VntRating } from './rating';
export { default as VntSelect } from './select';
export { default as VntSlider } from './slider';
export { default as VntToggle } from './toggle';
