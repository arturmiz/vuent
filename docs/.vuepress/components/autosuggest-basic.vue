<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries"></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestBasic',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
