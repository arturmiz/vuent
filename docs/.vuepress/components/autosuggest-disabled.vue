<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries" disabled></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestDisabled',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
