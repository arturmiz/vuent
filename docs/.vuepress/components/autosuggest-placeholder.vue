<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries"
                     placeholder="Type something..."></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestPlaceholder',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
