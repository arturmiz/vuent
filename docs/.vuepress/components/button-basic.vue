<template>
  <div class="playground">
    <vnt-button>Button</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonBasic',
  components: {
    VntButton
  }
}
</script>
