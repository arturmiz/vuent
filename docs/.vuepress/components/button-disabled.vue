<template>
  <div class="playground">
    <vnt-button disabled>Button</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonDisabled',
  components: {
    VntButton
  }
}
</script>
