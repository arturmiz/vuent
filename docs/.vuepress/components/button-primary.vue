<template>
  <div class="playground">
    <vnt-button primary>Primary</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonPrimary',
  components: {
    VntButton
  }
}
</script>
