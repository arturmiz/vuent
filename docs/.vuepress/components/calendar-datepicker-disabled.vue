<template>
  <div class="playground">
    <vnt-calendar-datepicker v-model="date" disabled />
    <div class="result">date: {{ date }}</div>
  </div>
</template>

<script>
import { VntCalendarDatepicker } from "../../../src/components";

export default {
  name: "CalendarDatepickerBasic",
  components: {
    VntCalendarDatepicker
  },
  data() {
    return {
      date: "2020-04-05"
    };
  }
};
</script>
