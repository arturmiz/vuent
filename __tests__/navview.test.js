import { createLocalVue, mount } from '@vue/test-utils';
import { VntNavView } from '@/components';
import { isInstalled } from './utils';

describe('NavView', () => {
  let wrapper;

  test('can be installed separately', () => {
    const localVue = createLocalVue();
    localVue.use(VntNavView);

    expect(isInstalled(localVue, VntNavView)).toBe(true);
  });

  describe('by default', () => {

    beforeAll(() => {
      wrapper = mount(VntNavView);
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

  describe('generates correct stucture', () => {

    beforeAll(() => {
      wrapper = mount(VntNavView, {
        slots: {
          default: `
            <vnt-navview pane-title="App name">
              <vnt-navview-navitems>

              </vnt-navview-navitems>
              <vnt-navview-content>
                Some content
              </vnt-navview-content>
            </vnt-navview>`
        }
      });
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

  describe('when using custom pane header', () => {

    beforeAll(() => {
      wrapper = mount(VntNavView, {
        slots: {
          default: `
            <vnt-navview pane-title="App name">
              <vnt-navview-pane-header>
                <div>App name with logo</div>
              </vnt-navview-pane-header>
              <vnt-navview-content>
                Some content
              </vnt-navview-content>
            </vnt-navview>`
        }
      });
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

});
