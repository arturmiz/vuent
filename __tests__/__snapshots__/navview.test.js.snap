// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavView by default renders correctly 1`] = `
<article class="vnt-navview">
  <aside class="vnt-navview__pane vnt-navview__pane--auto">
    <div class="vnt-navview__pane-header">
      <button type="button" class="vnt-navview__pane-toggle">
        <svg xmlns="http://www.w3.org/2000/svg" width="16" height="9" viewBox="0 0 16 9">
          <g transform="translate(-125.5 -80)">
            <line x2="16" transform="translate(125.5 80.5)" class="vnt-navview__pane-toggle-line"></line>
            <line x2="16" transform="translate(125.5 84.5)" class="vnt-navview__pane-toggle-line"></line>
            <line x2="16" transform="translate(125.5 88.5)" class="vnt-navview__pane-toggle-line"></line>
          </g>
        </svg>
      </button>
      <div class="vnt-navview__pane-title"></div>
    </div>
    <ul class="vnt-navview__items"></ul>
  </aside>
  <div class="vnt-navview__viewport">
    <!---->
    <section class="vnt-navview__content"></section>
  </div>
</article>
`;

exports[`NavView generates correct stucture renders correctly 1`] = `
<article class="vnt-navview">
  <aside class="vnt-navview__pane vnt-navview__pane--auto">
    <div class="vnt-navview__pane-header">
      <button type="button" class="vnt-navview__pane-toggle">
        <svg xmlns="http://www.w3.org/2000/svg" width="16" height="9" viewBox="0 0 16 9">
          <g transform="translate(-125.5 -80)">
            <line x2="16" transform="translate(125.5 80.5)" class="vnt-navview__pane-toggle-line"></line>
            <line x2="16" transform="translate(125.5 84.5)" class="vnt-navview__pane-toggle-line"></line>
            <line x2="16" transform="translate(125.5 88.5)" class="vnt-navview__pane-toggle-line"></line>
          </g>
        </svg>
      </button>
      <div class="vnt-navview__pane-title"></div>
    </div>
    <ul class="vnt-navview__items"></ul>
  </aside>
  <div class="vnt-navview__viewport">
    <!---->
    <section class="vnt-navview__content"></section>
  </div>
</article>
`;

exports[`NavView when using custom pane header renders correctly 1`] = `
<article class="vnt-navview">
  <aside class="vnt-navview__pane vnt-navview__pane--auto">
    <div class="vnt-navview__pane-header">
      <button type="button" class="vnt-navview__pane-toggle">
        <svg xmlns="http://www.w3.org/2000/svg" width="16" height="9" viewBox="0 0 16 9">
          <g transform="translate(-125.5 -80)">
            <line x2="16" transform="translate(125.5 80.5)" class="vnt-navview__pane-toggle-line"></line>
            <line x2="16" transform="translate(125.5 84.5)" class="vnt-navview__pane-toggle-line"></line>
            <line x2="16" transform="translate(125.5 88.5)" class="vnt-navview__pane-toggle-line"></line>
          </g>
        </svg>
      </button>
      <div class="vnt-navview__pane-title"></div>
    </div>
    <ul class="vnt-navview__items"></ul>
  </aside>
  <div class="vnt-navview__viewport">
    <!---->
    <section class="vnt-navview__content"></section>
  </div>
</article>
`;
